product_register.php

```diff
@@ -48,34 +39,19 @@
         }
     }
     
-    if (empty($error[$pic_name]) && $fields_validated){
-
+    if (empty($error[$pic_name]) && $fields_validated) {
+        $data = ['category'=>$_POST['category'], 'user_id'=> '1', 'name'=> $_POST['product_name'],
+                    'amount'=>$_POST['product_price'], 'description'=>$_POST['description'] ];
         if(!$is_update) {
-            $sql = "INSERT INTO `products_list` ( `category`,`user_id`, `name`, `amount`, "
-                    . "`description`, `created_date`) VALUES "
-                    . "( '" . $_POST['category'] . "','1','" . $_POST['product_name']
-                    . "', '" . $_POST['product_price'] . "', '" . $_POST['description']
-                    . "', NOW())";
-
+            $product_id = $db->insert_or_update(1, 'products_list', $data);
         } else {
-            $sql = "UPDATE `products_list` SET `category` = '". $_POST['category']
-                    . "', `name` = '" . $_POST['product_name'] 
-                    . "', `amount` = '". $_POST['product_price']
-                    . "', `description` = '". $_POST['description']
-                     . "' WHERE `products_list`.`id` = ". $_GET['update_id'];
-        }
-        if ( ! mysqli_query($conn, $sql)) {
-               $msg= "New record in PRODUCT_LIST FAILURE";
-               echo 'err1-->'. mysqli_error($conn);
-               header('Location: error.php');
-              // exit;
+            $db->insert_or_update(2,'products_list',$data,['id'=>$_GET['update_id']]);
+            $product_id = $_GET['update_id'];
         }
-
-        $product_id = $is_update ? $_GET['update_id'] : mysqli_insert_id($conn);
         
         if($is_update && $_FILES[$pic_name]['size'] != 0) {
-           $sql_get_image = "SELECT `image` from `products_list` WHERE `id`=".$_GET['update_id'];
-           $img_to_update = mysqli_fetch_assoc(mysqli_query($conn, $sql_get_image));
+            $db->select('products_list', ['image'],['id'=>$_GET['update_id']]);
```

Comment:
if($is_update && $_FILES[$pic_name]['size'] != 0) {
should be
if ($is_update && $_FILES[$pic_name]['size'] !== 0) {

space after if and use strict !== in place of !=